hack/update-apigen.sh

```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+set -e
+
+# this hash contains support for additionalProperties that we make use of, 0.9.0 does not contain it yet
+go install github.com/atombender/go-jsonschema/cmd/gojsonschema@v0.9.1-0.20210803111510-5167a44a3a0c
```

Comment:
unfortunately this package contains its main package in another package without anything else in there:
https://github.com/atombender/go-jsonschema/blob/master/cmd/gojsonschema/main.go

so when I want to vendor the main package, I get "'github.com/atombender/go-jsonschema/cmd/gojsonschema' is a program, not an importable package".

vendoring the "pkg", will not vendor the main.go sadly, so it's not invocable. Any ideas how to fix that?

Reply:
I kinda hacked it into the folder, but I'm sure it will disappear the next time we have to update the vendor dir. let me know if there's a working alternative somewhere.

Reply:
```
diff --git a/go.mod b/go.mod
index 7b9009d..3af986b 100644
--- a/go.mod
+++ b/go.mod
@@ -3,6 +3,7 @@ module github.com/openshift/must-gather-clean
 go 1.16
 
 require (
+	github.com/atombender/go-jsonschema v0.9.0
 	github.com/openshift/build-machinery-go v0.0.0-20210712174854-1bb7fd1518d3
 	github.com/spf13/cobra v1.1.3
 )
diff --git a/pkg/tools/tools.go b/pkg/tools/tools.go
index 72deb1f..bb2f7de 100644
--- a/pkg/tools/tools.go
+++ b/pkg/tools/tools.go
@@ -1,5 +1,6 @@
 package tools
 
 import (
+	_ "github.com/atombender/go-jsonschema/cmd/gojsonschema"
 	_ "github.com/openshift/build-machinery-go"
 )
```
Then run `go mod vendor`.

Reply:
> +	_ "github.com/atombender/go-jsonschema/cmd/gojsonschema"

how does that compile for you?

Reply:
Building the `tools` package is not possible. And since it's not imported anywhere it should not cause an import error.
```
❯ go version   
go version go1.16.6 linux/amd64
❯ make build
fatal: No names found, cannot describe anything.
go build -mod=vendor -trimpath -ldflags "-X github.com/openshift/must-gather-clean/pkg/version.versionFromGit="v0.0.0-unknown" -X github.com/openshift/must-gather-clean/pkg/version.commitFromGit="242a1a8" -X github.com/openshift/must-gather-clean/pkg/version.gitTreeState="dirty" -X github.com/openshift/must-gather-clean/pkg/version.buildDate="2021-08-04T07:02:19Z" " github.com/openshift/must-gather-clean/cmd/must-gather-clean
```

Reply:
You're right, it builds fine but "go vet" complains. Can we please add this to the CI build as well? I assume we have to ignore the tools package then.

>[tjungblu ~/git/must-gather-clean]$ make verify
Running `gofmt -s -l` on 6 file(s).
go vet -mod=vendor ./...
pkg/tools/tools.go:4:2: import "github.com/atombender/go-jsonschema/cmd/gojsonschema" is a program, not an importable package
make: *** [vendor/github.com/openshift/build-machinery-go/make/targets/golang/verify-update.mk:26: verify-govet] Error 1

Reply:
added the ignore and the right package for the CLI to be vendored.